tiledb/sm/array_schema/domain.h

```diff
@@ -733,6 +815,40 @@ class Domain {
   /** The tile order of the array the domain belongs to. */
   Layout tile_order_;
 
+  /**
+   * Vector of functions, one per dimension, for comparing the cell order of
+   * two coordinates. The inputs to the function are:
+   *
+   * - dim: The dimension to compare on.
+   * - coord_buffs: The coordinates, split in one buffer per dimensions.
+   * - a,b: The two positions of the coordinates to compare.
+   * - d: The dimension index to compare on.
+   */
+  std::vector<int (*)(
+      const Dimension* dim,
```

Comment:
How about `typedef` or `using =` either the vector its element template? Just an easy way to re-use it with the `tile_order_cmp_func_` declaration and any other place we may need to reuse it.